tensorboard/webapp/metrics/views/main_view/card_grid_component.ng.html

```diff
@@ -14,31 +14,31 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 -->
-<ng-container
-  *ngTemplateOutlet="groupControls; context: {isBottomControl: false}"
-></ng-container>
 
-<div class="card-grid">
-  <card-view
-    *ngFor="let item of cardIdsWithMetadata; trackBy: trackByCards"
-    [cardId]="item.cardId"
-    [groupName]="groupName"
-    [pluginType]="item.plugin"
-    [cardObserver]="cardObserver"
-    [cardLazyLoader]="item.cardId"
-  >
-  </card-view>
-</div>
+<div *ngIf="cardIdsWithMetadata.length >= 1">
```

Comment:
Is this absolutely required? If so, why?

Reply:
I don't want to render the paddings when there is no cards in a group.

Reply:
I guess the part that is not resonating with me is the fact that you are using `cardIdWithMetadata.length` to determine whether the group is expanded or not. Group expansion is something that is different than cardIdWithMetadata so I wish to see explicit state pass down and render accordingly. I think that will be more future proof to a world where cardIdwithMetadata can be legal be empty while card is expanded.